ui/src/app/guards/access-tasks-guard.service.ts

```diff
@@ -4,23 +4,32 @@ import {
   RouterStateSnapshot
 } from '@angular/router';
 import {ProfileStorageService} from 'app/services/profile-storage.service';
+import {ServerConfigService} from 'app/services/server-config.service';
 import {Observable} from 'rxjs/Observable';
 
 @Injectable()
-export class EraCommonGuard implements CanActivate, CanActivateChild {
+export class AccessTasksGuard implements CanActivate, CanActivateChild {
 
-  constructor(private router: Router, private profileStorageService: ProfileStorageService) {}
+  constructor(
+    private router: Router,
+    private profileStorageService: ProfileStorageService,
+    private serverConfigService: ServerConfigService,
+  ) {}
 
   canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot):
   Observable<boolean> {
     if (route.routeConfig.path === '' ||
-        route.routeConfig.path.startsWith('nih-callback/')) {
-      // Leave /admin unguarded in order to allow bootstrapping of verified users.
+        route.routeConfig.path.startsWith('nih-callback')) {
       return Observable.from([true]);
     }
+    let enforceRegistered: boolean;
+    this.serverConfigService.getConfig().subscribe((config) => {
```

Comment:
These callbacks (here, and the profile call) are likely, but not guaranteed to execute in the order you are expecting here. Should wrap one of these callbacks in the other (or chain with flatmap)